### ISSUE TITLE
About conjugate matrix trace

### ISSUE THREAD

**Issue by milkway124** [symengine/symengine#1723]
Hi, 

I think I face an error in "conjugate & eval_complex_double".

I test matrix conjugate and transpose, but I have a problem.
: "terminate called after throwing an instance of 'SymEngine::NotImplementedError'
 what(): Not Implemented"

the example matrix A is 
[(0.0 + 1.0*I)*sin(2.0*x) + cos((0.8 + 1.6*I)*x), 0]
[0, 0.0 + 1.0*I + cos((0.2 - 5.6*I)*x)]

and then, I use "conjA.trace;",
`conjugate_transpose_dense(A,conjA); `

 "conjugate_transpose_dense(A,conj);", "subs" and eval_complex_double".
`    
    RCP<const Basic> ATrace;
    ATrace=conjA.trace();
    std::cout << *ATrace << std::endl;
`

    for(int j=0; j<10;j++){
        sub_val[x]=real_double(j);
        Result=ATrace->subs(sub_val);
        NumResult = eval_complex_double(*Result);
        std::cout << sub_val << NumResult << std::endl;
    }

At that point, I have the "what()" problem.

To solve this problem, I use a different method that is using conjugate directly.
`    conjA.set(0,0, 
        add(cos(mul(mul(conjugate(x),real_double(2)),complex_double(doublec(0.4,- 0.8)))),mul(complex_double(doublec(0,-1)), sin(mul(conjugate(x),real_double(2)))))
        //cos(2*conjx*[0.4,-0.8I]) + [0,-I]*sin(2*conjx)
    );
    conjA.set(1,1,
        add(cos(mul(mul(conjugate(x),real_double(2)),complex_double(doublec(-0.1, -2.8)))),complex_double(doublec(0,-1))) 
        //cos(2*conjx*[-0.1,-2.8I]) + [0,-I]
    );
`
:[(0.0 - 1.0*I)*sin(2.0*conjugate(x)) + cos((0.8 - 1.6*I)*conjugate(x)), 0]
[0, 0.0 - 1.0*I + cos((0.2 + 5.6*I)*conjugate(x))]

using above the matrix, I can evaluate the conjugate matrix trace.

I do not want to use the direct method, How to fix it? 
Please let me know it.

Thank you.

**Comment by rikardn** [comment]
It is a bit hard to follow what your problem is. Could you please post a minimal example program that can reproduce the problem.

**Comment by milkway124** [comment]
Ok, I upload the example code about conjugate.

[conjugate_err.zip](https://github.com/symengine/symengine/files/5903104/conjugate_err.zip)

I face the error using the first conjugate method. 

Thank you.

**Comment by rikardn** [comment]
Sorry for being picky, but to maximize your chances of someone taking a look at your problem I would advice you to give a minimal reproducible example inline in the issue report together with a description of the problem. Make sure that there is nothing more or nothing less needed to reproduce the problem. An example:


The code below crash at line 3 with a division by zero error. I expected it to return `Inf`.

```c++
int main()
{
    return 1 / 0;
}
```

**Comment by milkway124** [comment]
I am sorry to have bothered you. 

I try to calculate the conjugate matrix, but I meet an Error; "terminate called after throwing an instance of SymEngine::NotImplementedError' what():  Not Implemented"

So, I want to get some advice on this problem.

In code, I want to get a trace of the conjugate transpose A matrix.

First, I prepare the A matrix-like below, (at line 41~46)
2x2 matrix A ={ cos(2*x*[0.4,-0.8I])+[0,-I]*sin(2*x),   0,
                          0 ,          cos(2*x*[-0.1,-2.8I])+[0,-I])   }
(x : symbol, I : Imaginary number)

and then, I try to conjugate, transpose, and a trace of A matrix at line 58

Finally, I try to substitute x to a real double type number at line 82

I expect that I can get the number, but I just get the error; "terminate called after throwing an instance of SymEngine::NotImplementedError' what():  Not Implemented"

Thank you.
